File: java/src/main/java/io/fivexer/sdk/model/WorkerSessionToken.java
package io.fivexer.sdk.model;

/**
 * A {@code wt_} bearer token scoped to exactly one worker.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerSessionToken {
    private String token;
    private String expiresAt;

    public String getToken() { return token; }

    /**
     * ISO-8601 moment this token stops working, or null on a server predating the refresh
     * endpoint. Null rather than a defaulted timestamp: a made-up expiry would rotate either far
     * too early or never, and rotating only on a 401 puts a failed request in front of a person.
     */
    public String getExpiresAt() { return expiresAt; }
}

File: php/src/Model/WorkerSessionToken.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A `wt_` bearer token scoped to exactly one worker.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerSessionToken
{
    public function __construct(
        public readonly string $token,
        /**
         * ISO-8601 moment this token stops working, or null on a server predating the refresh
         * endpoint. Null rather than a defaulted timestamp: a made-up expiry would rotate either
         * far too early or never, and rotating only on a 401 puts a failed request in front of
         * a person at the start of every session.
         */
        public readonly ?string $expiresAt = null,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            token: (string) ($data['token'] ?? ''),
            expiresAt: isset($data['expiresAt']) ? (string) $data['expiresAt'] : null,
        );
    }
}
